include/engine/routing_algorithms.hpp

```diff
@@ -57,7 +57,16 @@ class RoutingAlgorithmsInterface
 template <typename Algorithm> class RoutingAlgorithms final : public RoutingAlgorithmsInterface
 {
   public:
-    RoutingAlgorithms(SearchEngineData<Algorithm> &heaps,
+    // HeapAlgorithm is an explicit heap algorithm type to be used with Algorithm
+    // - CH algorithms use CH heap
+    // - CoreCH algorithms use CH heap
+    // - MLD algorithms use MLD heap
+    using HeapAlgorithm = typename std::conditional<
+        std::is_same<Algorithm, routing_algorithms::corech::Algorithm>::value,
+        routing_algorithms::ch::Algorithm,
+        Algorithm>::type;
```

Comment:
Good idea! But i don't know what exactly to check: `HeapAlgorithm` is a derived type and not a parameter, so static asserts with `HeapAlgorithm` will just check `std::conditional` implementation. Also function definitions have explicit types, so checks on that side will be useless.

I would like to have a human-readable compiler message, at the moment incorrect types will lead to messages like " bla   **has not been declared.**  bla   **note: suggested alternatives:** bla bla bla "